app/policies/repository_policy.rb

```diff
@@ -20,7 +20,12 @@ def show?
   # Returns true if the repository can be destroyed.
   def destroy?
     raise Pundit::NotAuthorizedError, "must be logged in" unless @user
-    @user.admin? || @repository.namespace.team.owners.exists?(user.id)
+
+    delete_enabled         = APP_CONFIG.enabled?("delete")
+    is_owner               = @repository.namespace.team.owners.exists?(user.id)
+    can_contributor_delete = APP_CONFIG["delete"]["contributors"] &&
```

Comment:
Change to: `APP_CONFIG.enabled?("delete.contributors")` as you are doing in the tag policy. Moreover, maybe it would be a good change to DRY this part and create a method like this:

```ruby
def allow_destroy_for?(user, repository)
  delete_enabled         = APP_CONFIG.enabled?("delete")
  is_owner               = repository.namespace.team.owners.exists?(user.id)
  can_contributor_delete = APP_CONFIG.enabled?("delete.contributors") &&
                             @repository.namespace.team.contributors.exists?(user.id)
  delete_enabled && (user.admin? || is_owner || can_contributor_delete)
end
```

Note that the tag policy basically does `tag.repository`, so the repository is the only object needed.

Reply:
> Change to: APP_CONFIG.enabled?("delete.contributors") as you are doing in the tag policy.

The tag policy was actually a leftover since it doesn't work. The `enabled?` function works only with the `enabled` child attribute. In this case we have `contributors` being a child of `delete` and the value goes directly to it and not to a `enabled` child. I should actually fix the tag policy file.

I double checked and by using `enabled?` on `delete.contributors`, it always return `true`. I think that if `enabled` child attribute does not exist it should return false, no?